frontend/src/screens/SigninScreen.js

```diff
@@ -1,19 +1,39 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
 import { Link } from 'react-router-dom';
+import { signin } from '../actions/userActions';
+import LoadingBox from '../components/LoadingBox';
+import MessageBox from '../components/MessageBox';
 
-export default function SigninScreen() {
+export default function SigninScreen(props) {
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
+
+  const redirect = props.location.search
```

Comment:
In react-router V6 the way to achieve this is by using 
`import { Link, useNavigate } from "react-router-dom";

const submitHandler = (e) => {
    /**
     * This will make sure that the form will not refresh
     * Will make use of ajax request to perform
     */

    e.preventDefault();
    dispatch(signin(email, password));
    if (!error) {
      navigate("/");
    }
  };

`

Reply:
This will not redirect you to /shipping page, if you click "proceed to checkout", when you were not signed in initially.
`const navigation = useRef(useNavigate());
 const { search } = useLocation();
 const searchSplit = search.split('=')[1];
 const redirect = search ? `/${searchSplit}` : '/';`

useEffect(() => {
        if (userInfo) {
            navigation.current(redirect);
        }
    }, [userInfo, navigation, redirect])

Reply:
just 
import { useNavigate } from "react-router-dom";

befor return
    let navigate = useNavigate();

useEffect(() => {
        if(userInfo){
            navigate("/shiping")
        }
    }, [userInfo])
that will redirect you to "/shiping"

Reply:
in my case is not working props.location or any case so what I have done correctly?
 first go to cart screen and this is our checkouthandler  const checkoutHandler = () => {
    navigation(`/signin?redirect=shipping`);
  };

so what i did to redirect my page const checkoutHandler = () => {
    if (userInfo) {
      navigation("/shipping");
    } else {
      navigation("/signin");
    }
  };

and go to signinscreen and set the navigation and get the data
const navigate = useNavigate();
  const userSignIn = useSelector((state) => state.userSignIn);
  const { loading, error, userInfo } = userSignIn; 
 useEffect(() => {
    if (userInfo) {
      navigate("/");
    }
  }, [userInfo]);
and redirect user like this and works good and your this auth successfully work
thank you basir sir you are one of my very good teacher